n3fit/src/evolven3fit_new/utils.py

```diff
@@ -178,15 +178,17 @@ def get_theoryID_from_runcard(usr_path):
     return my_runcard["theory"]["theoryid"]
 
 
-def generate_q2grid(Q0, Qfin, Q_points, match_dict, nf0=None):
+def generate_q2grid(Q0, Qfin, Q_points, match_dict, nf0=None, ic=False):
     """Generate the q2grid used in the final evolved pdfs or use the default grid if Qfin or Q_points is
     not provided.
 
     match_dict contains the couples (mass : factor) where factor is the number to be multiplied to mass
     in order to obtain the relative matching scale.
     """
     if Qfin is None and Q_points is None:
-        if nf0 == 4:
+        if nf0 == 4 and ic:
```

Comment:
```suggestion
        if nf0 == 3 or ic:
```
this way you can remove the last branch.